src/DataCollector/PheanstalkDataCollector.php

```diff
@@ -50,31 +52,26 @@ public function reset()
     /**
      * @inheritdoc
      */
-    public function collect(Request $request, Response $response, \Exception $exception = null)
+    public function collect(Request $request, Response $response, Exception $exception = null)
     {
         $defaultPheanstalk = $this->pheanstalkLocator->getDefaultPheanstalk();
 
         // Collect the information
+        /** @var PheanstalkInterface $pheanstalk */
         foreach ($this->pheanstalkLocator->getPheanstalks() as $name => $pheanstalk) {
             // Get information about this connection
             $this->data['pheanstalks'][$name] = [
-                'name'      => $name,
-                'host'      => $pheanstalk->getConnection()->getHost(),
-                'port'      => $pheanstalk->getConnection()->getPort(),
-                'timeout'   => $pheanstalk->getConnection()->getConnectTimeout(),
-                'default'   => $defaultPheanstalk === $pheanstalk,
-                'stats'     => [],
-                'listening' => $pheanstalk->getConnection()->isServiceListening(),
+                'name'    => $name,
+                'default' => $defaultPheanstalk === $pheanstalk,
+                'stats'   => [],
             ];
 
-            // If connection is not listening, there is a connection problem.
-            // Skip next steps which require an established connection
-            if (!$pheanstalk->getConnection()->isServiceListening()) {
-                continue;
+            try {
+                $pheanstalkStatistics = iterator_to_array($pheanstalk->stats());
+            } catch (ConnectionException $exception) {
+                $pheanstalkStatistics = null;
```

Comment:
This should be `continue`. If `stats()` fails with a `ConnectionException` the following pheanstalk operations will (probably) fail as well. As of now line 82 (`$pheanstalkStatistics['current-jobs-ready']`) would fail, anyway.